backend/app/services/llm.py

```
@@ -345,12 +345,86 @@ def _extract_explicit_topic(question: str) -> Optional[str]:
+
+    topic = _clean_suggestion_text(raw_topic or raw_query or "")
+    query = _clean_suggestion_text(raw_query or raw_topic or "")
+    if not topic:
+        return None
+
+    if len(topic.split()) > MAX_SUGGESTION_WORDS:
+        return None
+    if len(query.split()) > MAX_SUGGESTION_WORDS:
+        query = topic
+
+    normalized_topic = _normalize_text(topic)
```

Ruff (0.15.11)

[error] Undefined name `_normalize_text`

(F821)